File: Spring/Task 9/Task 2.9/Command.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bash
{
    public abstract class Command
    {
        public List<string> Arguments { get; set; }

        public abstract void Execute();

        public abstract List<string> ExecutePipe();
    }
}

File: Spring/task9/src/Commands/Command.java
package Commands;

import Bash.*;

public abstract class Command {

    protected ClientInterface clientInterface;
    protected String directoryName;
    protected Bash bash;

    public Command(ClientInterface clientInterface, String directoryName, Bash bash) {
        this.directoryName = directoryName;
        this.clientInterface = clientInterface;
        this.bash = bash;
    }

    public abstract void run(String[] args, String[] commands);
}